flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/DefaultSchemaResolver.java

```diff
@@ -226,11 +225,10 @@ private void validateTimeColumn(String columnName, List<Column> columns) {
     }
 
     private void validateWatermarkExpression(LogicalType watermarkType) {
-        if (!hasRoot(watermarkType, LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE)
-                || getPrecision(watermarkType) != 3) {
+        if (!supportedWatermarkType(watermarkType) || getPrecision(watermarkType) != 3) {
```

Comment:
Please also remember to validate watermark output type should be the same with rowtime type. And please add tests for this.